account-kit/core/src/store/store.ts

```diff
@@ -94,13 +97,22 @@ export const createAccountKitStore = (
                 return createInitialStoreState(params);
               }
 
-              // this is the default merge behavior
-              return { ...current, ...persistedState };
+              const { chain, ...connection } = persistedConnections.find(
+                (x) => x.chain.id === persistedState.bundlerClient.chain.id
```

Comment:
it's fine to use the persisted state here because we've already established the persisted client uses the same connections as the core state. this way if a user has switched chains that switch persists (see tests above)